src/target.cpp

```diff
@@ -78,22 +78,22 @@ namespace powerloader
         }
     }
 
-    CbReturnCode Target::call_endcallback(TransferStatus status)
+    CbReturnCode Target::call_end_callback(TransferStatus status)
     {
         reset_file(status);
 
-        end_callback end_cb = override_endcb ? override_endcb : target->endcb;
-        void* cb_data = override_endcb ? override_endcb_data : target->cbdata;
+        end_callback_t end_cb
+            = override_end_callback ? override_end_callback : target->end_callback;
```

Comment:
Potential minor improvement: if either `override_end_callback` or `target->end_callback` are not needed anymore after being called, moving them would be potentially less expensive.

Reply:
hmm, not sure if moving works, but we could take a reference or a pointer?

Reply:
A reference would work fine here yes.

Reply:
I made the left hand side a reference.